rpc/provider_test.go

```diff
@@ -229,12 +118,17 @@ func TestCookieManagement(t *testing.T) {
 				Path:  "/",
 			})
 		} else {
-			var result string
-			err := mock_starknet_chainId(&result)
+			var rawResp json.RawMessage
+			err := mock_starknet_chainId(&rawResp)
 			if err != nil {
 				http.Error(w, err.Error(), http.StatusInternalServerError)
 				return
 			}
+			var result string
+			if err := json.Unmarshal(rawResp, &result); err != nil {
+				http.Error(w, err.Error(), http.StatusInternalServerError)
+				return
+			}
```

Comment:
Just curious, why this change?

Reply:
The test was failing.
mock_chainId now requires a raw json var as input and this test was passing a string directly to it
https://github.com/NethermindEth/starknet.go/pull/619/files#diff-617ac3083426917993255a7947979919a2700fce3d4d158586d30561ee91e396R164

Reply:
Usually, all other methods call mock methods through "do()" function, and this function calls the methods passing json.RawMessage, that's why all of them passed and this one not